ec2-config/instance.tf

```diff
@@ -0,0 +1,93 @@
+# Configure the S3 backend
+terraform {
+  backend "s3" {
+      bucket = "terraform-state-bucket-180122"
+      key = "global/s3/ec2/terraform.tfstate"
+      dynamodb_table = "terraform-state-lock"
+      region = "us-east-1"
+      encrypt = true
+  }
+}
+
+# Configure the terraform block
+terraform {
+  required_providers {
+      aws = {
+          source = "hashicorp/aws"
+          version = "~> 3.0"
+      }
+  }
+
+  required_version = ">= 0.14.9"
+}
+
+# Configure the provider
+provider "aws" {
+  region = "us-east-1"
+}
+
+# Filter the aws ami
+data "aws_ami" "ubuntu" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"] # Canonical
+}
+
+# Define the module
+module "network" {
+    source = "../network-config"   
+}
```

Comment:
Please don't use modules for now. Instead, create a `network.tf` file in the same folder.
We will cover modules a little later Otherwise it will become confusing